bitacross-worker/bitacross/core/bc-task-processor/src/lib.rs

```diff
@@ -0,0 +1,563 @@
+}
+
+#[allow(clippy::type_complexity)]
+fn handle_ceremony_command<SKR, SIGNINGAK, EKR, BKR, S, H, O, RRL, ERL, SRL, Responder>(
+	context: Arc<BitAcrossTaskContext<SKR, SIGNINGAK, EKR, BKR, S, H, O, RRL, ERL, SRL, Responder>>,
+	ceremony_id: CeremonyId,
+	command: CeremonyCommand,
+	event_threads_pool: ThreadPool,
+	peers_map: Arc<Mutex<HashMap<[u8; 32], DirectRpcClient>>>,
+) where
+	SKR: AccessKey + AccessPubkey<KeyType = Rsa3072PubKey> + Send + Sync + 'static,
+	SIGNINGAK: AccessKey<KeyType = ed25519::Pair> + Send + Sync + 'static,
+	EKR: AccessKey<KeyType = EcdsaPair> + Send + Sync + 'static,
+	BKR: AccessKey<KeyType = SchnorrPair> + Send + Sync + 'static,
+	<SKR as AccessKey>::KeyType: ShieldingCryptoEncrypt + ShieldingCryptoDecrypt + 'static,
+	S: StfEnclaveSigning<TrustedCallSigned> + Send + Sync + 'static,
+	H: HandleState + Send + Sync + 'static,
+	H::StateT: SgxExternalitiesTrait,
+	O: EnclaveOnChainOCallApi + EnclaveMetricsOCallApi + EnclaveAttestationOCallApi + 'static,
+	RRL: RelayerRegistryLookup + Send + Sync + 'static,
+	ERL: EnclaveRegistryLookup + Send + Sync + 'static,
+	SRL: SignerRegistryLookup + Send + Sync + 'static,
+	Responder: SendRpcResponse<Hash = H256> + Send + Sync + 'static,
+{
+	// check whether to store command to tmp
+	let is_first_round = {
+		context
+			.ceremony_registry
+			.read()
+			.unwrap()
+			.get(&ceremony_id)
+			.map(|(c, _)| c.read().unwrap().is_first_round())
+	};
+	match (is_first_round, &command) {
+		(None, CeremonyCommand::SaveNonce(_, _))
+		| (Some(true), CeremonyCommand::SavePartialSignature(_, _)) => {
+			context
+				.ceremony_command_tmp
+				.write()
+				.unwrap()
+				.entry(ceremony_id)
+				.and_modify(|(command_tmp, _)| command_tmp.write().unwrap().push(command.clone()))
+				.or_insert((Arc::new(RwLock::new(vec![command])), get_current_timestamp()));
+			return
+		},
+		(Some(true), CeremonyCommand::SaveNonce(_, _))
+		| (Some(true), CeremonyCommand::Init)
+		| (Some(false), CeremonyCommand::SavePartialSignature(_, _)) => {},
+		(is_first_round, command) => {
+			error!(
+				"receive wrong command: is_first_round: {:?}, command: {:?}, drop it",
+				is_first_round, command
+			);
+			return
+		},
+	}
+
+	// try to udpate peers_map
+	let my_identity: Address32 =
+		context.signing_key_access.retrieve_key().unwrap().public().0.into();
+	context
+		.enclave_registry_lookup
+		.get_all()
+		.iter()
+		.for_each(|(identity, address)| {
+			if my_identity != *identity
+				&& !peers_map.lock().unwrap().contains_key(identity.as_ref())
+			{
+				info!("creating new connection to peer: {:?}", address);
+				match (DirectRpcClientFactory {}).create(address) {
+					Ok(client) => {
+						peers_map.lock().unwrap().insert(*identity.as_ref(), client);
+					},
+					Err(e) => error!("Could not connect to peer {}, reason: {:?}", address, e),
+				}
+			}
+		});
+
+	// process commands and events
+	let mut commands_to_process = vec![command];
```

Comment:
if there is always single command to process, simplify this

Reply:
`ceremony_command_tmp` stores commands that not ready to be processed when they're received, so lenth of `commands_to_process` could be more than one